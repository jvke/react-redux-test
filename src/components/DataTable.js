import React from 'react';
import styled from 'styled-components';
import { lighten, transparentize, darken } from 'polished';


const DataTable = ({ children, ...otherProps }) => (
  <table {...otherProps}>
    {children}
  </table>
);

export default styled(DataTable)`
  & {
    margin: 0 auto;
    border-collapse: collapse;
    border: solid 1px ${props => lighten(0.5, props.theme.brand.light)};
    border-radius: 4px;
    background: ${props => transparentize(0.7, lighten(0.2, props.theme.brand.light))};
    padding: 10px;

    th {
      background: ${props => transparentize(0.5, darken(0.1, props.theme.brand.light))};
    }
    td, th {
      padding: 5px;
    }
    tr td {
      border-top: solid 1px ${props => lighten(0.5, props.theme.brand.light)};
    }
  }

`
