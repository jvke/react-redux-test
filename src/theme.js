export default {
  brand: {
    light: 'rgb(8, 62, 101)',
    dark: 'rgb(1, 20, 46)',
  },
  alert: {
    error: '#FF0000'
  },
  textColor: '#EEEEEE'
}
