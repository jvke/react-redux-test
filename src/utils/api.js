import { put, call } from 'redux-saga/effects';
const handleSagaAPICall = function* _handleSagaAPICall(fetchFn, successFn, failFn, endpoint, ...args) {
  try {
    // do axios call
    const results = yield call(fetchFn, endpoint, ...args);

    if (results.statusText === 'OK') {
      yield put(successFn(results.data))
    } else {
      throw new Error('Unknown error has occurred');
    }
  } catch (e) {
    if (e.response && e.response.data) {
      yield put(failFn(new Error(e.response.data.message)));
    } else {
      yield put(failFn(new Error(e.message)));
    }
  }
}

export { handleSagaAPICall };
