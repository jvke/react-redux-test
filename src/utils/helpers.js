import { ThemeProvider } from 'styled-components';
import React from 'react';
const wrapWithTheme = (theme) => {
  return (children) => (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  )
}

export { wrapWithTheme }
