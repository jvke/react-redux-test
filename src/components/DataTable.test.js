import React from 'react';
import ReactDOM from 'react-dom';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import DataTable from './DataTable';
import theme from '../theme';
import { wrapWithTheme } from '../utils/helpers';

import { ThemeProvider } from 'styled-components';

const wrapper = wrapWithTheme(theme);

describe('DataTable', () => {
  it('DataTable renders with no props specified', () => {
    const tree = renderer
      .create(
        <DataTable theme={theme}>
          <tbody>
            <tr>
              <td>Foo</td>
              <td>Bar</td>
            </tr>
          </tbody>
        </DataTable>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('DataTable renders with props specified', () => {
    const tree = renderer
      .create(<DataTable theme={theme} a={1} b={2} c={3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
