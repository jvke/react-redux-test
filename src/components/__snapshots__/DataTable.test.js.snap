// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable DataTable renders with no props specified 1`] = `
.c0 {
  margin: 0 auto;
  border-collapse: collapse;
  border: solid 1px #78c0f4;
  border-radius: 4px;
  background: rgba(15,120,196,0.3);
  padding: 10px;
}

.c0 th {
  background: rgba(4,33,54,0.5);
}

.c0 td,
.c0 th {
  padding: 5px;
}

.c0 tr td {
  border-top: solid 1px #78c0f4;
}

<table
  className="c0"
  theme={
    Object {
      "alert": Object {
        "error": "#FF0000",
      },
      "brand": Object {
        "dark": "rgb(1, 20, 46)",
        "light": "rgb(8, 62, 101)",
      },
      "textColor": "#EEEEEE",
    }
  }
>
  <tbody>
    <tr>
      <td>
        Foo
      </td>
      <td>
        Bar
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`DataTable DataTable renders with props specified 1`] = `
.c0 {
  margin: 0 auto;
  border-collapse: collapse;
  border: solid 1px #78c0f4;
  border-radius: 4px;
  background: rgba(15,120,196,0.3);
  padding: 10px;
}

.c0 th {
  background: rgba(4,33,54,0.5);
}

.c0 td,
.c0 th {
  padding: 5px;
}

.c0 tr td {
  border-top: solid 1px #78c0f4;
}

<table
  a={1}
  b={2}
  c={3}
  className="c0"
  theme={
    Object {
      "alert": Object {
        "error": "#FF0000",
      },
      "brand": Object {
        "dark": "rgb(1, 20, 46)",
        "light": "rgb(8, 62, 101)",
      },
      "textColor": "#EEEEEE",
    }
  }
/>
`;
